orchagent/copporch.cpp

```diff
@@ -0,0 +1,488 @@
+    {"mld_v2_done", SAI_HOSTIF_TRAP_ID_IPV6_MLD_V1_DONE},
+    {"mld_v2_report", SAI_HOSTIF_TRAP_ID_MLD_V2_REPORT},
+    {"ip2me", SAI_HOSTIF_TRAP_ID_IP2ME},
+    {"ssh", SAI_HOSTIF_TRAP_ID_SSH},
+    {"snmp", SAI_HOSTIF_TRAP_ID_SNMP},
+    {"router_custom_range", SAI_HOSTIF_TRAP_ID_ROUTER_CUSTOM_RANGE_BASE},
+    {"l3_mtu_error", SAI_HOSTIF_TRAP_ID_L3_MTU_ERROR},
+    {"ttl", SAI_HOSTIF_TRAP_ID_TTL_ERROR}
+};
+
+std::map<string, sai_packet_action_t> packet_action_map = {
+    {"drop", SAI_PACKET_ACTION_DROP},
+    {"forward", SAI_PACKET_ACTION_FORWARD},
+    {"copy", SAI_PACKET_ACTION_COPY},
+    {"copy_cancel", SAI_PACKET_ACTION_COPY_CANCEL},
+    {"trap", SAI_PACKET_ACTION_TRAP},
+    {"log", SAI_PACKET_ACTION_LOG},
+    {"deny", SAI_PACKET_ACTION_DENY},
+    {"transit", SAI_PACKET_ACTION_TRANSIT}
+};
+
+CoppOrch::CoppOrch(DBConnector *db, string tableName) :
+    Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+};
+
+bool CoppOrch::getTrapID(string &trap_id_str, sai_hostif_trap_id_t &trap_id) const
+{
+    SWSS_LOG_ENTER();
+    auto it = trap_id_map.find(trap_id_str);
+    if (it == trap_id_map.end())
+    {
+        SWSS_LOG_ERROR("Invalid trap_id specified %s\n", trap_id_str.c_str());
+        return false;
+    }
+    trap_id = trap_id_map[trap_id_str];
+    SWSS_LOG_DEBUG("Output value:%d", trap_id);
+    return true;
+}
+
+bool CoppOrch::getTrapIdList(vector<string> &trap_id_name_list, vector<sai_hostif_trap_id_t> &trap_id_list) const
+{
+    SWSS_LOG_ENTER();
+    for(auto trap_id_str : trap_id_name_list)
+    {
+        sai_hostif_trap_id_t trap_id;
+        if (!getTrapID(trap_id_str, trap_id))
+        {
+            SWSS_LOG_ERROR("Invalid trap_id value specified %s\n", trap_id_str.c_str());
+            return false;
+        }
+        SWSS_LOG_DEBUG("Input trap_id:%s, pushing trap_id:%d", trap_id_str.c_str(), trap_id);
+        trap_id_list.push_back(trap_id);
+    }
+    return true;
+}
+
+bool CoppOrch::applyTrapIds(sai_object_id_t trap_group, vector<string> &trap_id_name_list, std::vector<sai_attribute_t> &trap_id_attribs)
+{
+    SWSS_LOG_ENTER();
+    sai_status_t sai_status;
+    vector<sai_hostif_trap_id_t> trap_id_list;
+    if (!getTrapIdList(trap_id_name_list, trap_id_list))
+    {
+        return false;
+    }
+
+    sai_attribute_t attr;
+    attr.id = SAI_HOSTIF_TRAP_ATTR_TRAP_GROUP;
+    attr.value.oid = trap_group;
+    trap_id_attribs.push_back(attr);
+
+    // The channel is always the NETDEV device.
+    attr.id = SAI_HOSTIF_TRAP_ATTR_TRAP_CHANNEL;
+    attr.value.s32 = SAI_HOSTIF_TRAP_CHANNEL_NETDEV;
+    trap_id_attribs.push_back(attr);
+
+    for (auto trap_id : trap_id_list)
+    {
+        SWSS_LOG_DEBUG("set trap attr.1:%d", attr.id);
+        for(auto trap_id_attr : trap_id_attribs)
+        {
+            SWSS_LOG_DEBUG("set trap attr.2:%d", trap_id_attr.id);
+            if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->set_trap_attribute(trap_id, &trap_id_attr)))
```

Comment:
Could you align the previous code as to split this into two lines: one for the function call and another for the if. Same as below.